### DISCUSSION TITLE
S3 Bucket

### DISCUSSION THREAD

**Discussion by elmortero1** [felddy/foundryvtt-docker#328] [General]
Thank you for your container! You have made it so much easier to get my server up and running. I have a question regarding integrating my s3 bucket in to my settings. Is this a command I can put in the docker-compose file? Or is this something that would go in my foundry data settings?

**Comment by felddy** [comment]
Take a look at the `FOUNDRY_AWS_CONFIG` environment variable for the container: https://github.com/felddy/foundryvtt-docker#optional-variables

The value you assign to `FOUNDRY_AWS_CONFIG` will be applied to your `options.json` at container startup.

See Foundry's documentation here: https://foundryvtt.com/article/aws-s3/
> The aws field of the `options.json` file accepts either an in-line AWS configuration or a relative file path to a separate AWS configuration JSON file.

**Reply by MaddisonM79** [reply]
Following up on this. I have the file in the same directory as my docker-compose.yaml file. I get the error: [ServerConfiguration.awsConfig]: File path "/data/Config/vols/foundryvtt/Config/aws.json" does not exist

My docker-compose is:
version: "3.8"

services:
  foundry:
    image: felddy/foundryvtt:release
    hostname: abc.123.com
    init: true
    restart: always
    volumes:
      - type: bind
        source: ./vols/foundrydata
        target: /data
    environment:
      - FOUNDRY_PASSWORD=XXXXXXXX
      - FOUNDRY_USERNAME=XXXXXXXX
      - FOUNDRY_ADMIN_KEY=XXXXXXXX
      - FOUNDRY_LICENSE_KEY=XXXXXXX
      - FOUNDRY_AWS_CONFIG=aws.json
    ports:
      - target: 30000
        published: 30000
        protocol: tcp

I am sure I am missing something stupid. Can you help point me in the right direction?

**Reply by MaddisonM79** [reply]
Please disregard. I put the file int he right location, in the vols folder like the error said and it worked. Its been a long day of moving many different docker containers to one server and building a reverse proxy. :(. Works like a champ now!
